### DISCUSSION TITLE
Idea for bottle manometer based on iSpindel

### DISCUSSION THREAD

**Discussion by ralfsteck** [universam1/iSpindel#612] [Ideas]
Hi all,

I'm in the process of soldering a litte project that I call iBottle - a digital bottle manometer. I based it on a very similar hardware to the iSpindle - Wemos D1, DS18B20 for temperature, TP4056 charge controller and 18650 battery. Instead of a tilt unit I add a [pressure sensor](https://www.amazon.de/dp/B07T5BVV1D) and a 0.66" OLED display.

I plan to use ESPeasy to drive this setup (there is already a [similar project](https://www.letscontrolit.com/wiki/index.php?title=Analog_Pressure_sensor)), but it came to my mind if it would be possible to tweak the  iSpindle firmware to drive this (minus the OLED display, of course) and ideally connect this to the RasPi Server so it is possible to monitor both phases of the fermentation.

Is that totally nuts or what do you think?

**Comment by thegreatgunbantoad** [comment]
If you use one of those petling tubes you won't get much of a response from the pressure sensor as the tubes just don't deform enough to give a true reading.  They become an almost reverse pressure vessel.

I do have code for driving a pressure sensor (BMP280) that did work, but found it wasn't worth adding one to an iSpindle in the manner that I did.  I'll try to remember to upload it to git hub after I get Halloween done with.

**Comment by ralfsteck** [comment]
I never talked about a petling. It's a seperate project for bottle fermentation, not for main fermentation. It connects to the top of the bottle via an Adaptor (the big silver thing in the pic). I already have a breadboard version up & running.
![20231027_191522](https://github.com/universam1/iSpindel/assets/19465139/a721d61b-d692-4a28-a730-dbb94cbc7133)

**Reply by thegreatgunbantoad** [reply]
Oh nice job.  I don't think the BMP280 code is going to be of any value to you then.  I do still need to put it up on my github pages though.

**Reply by thegreatgunbantoad** [reply]
The code I used is here, but it was on an old version.

**Reply by ErikdBr** [reply]
How do you put that adapter in the bottle and how does it stay in when pressure builds up? I only see a metal head with screw thread. (I am interested to build one for myself :-) )

**Comment by ralfsteck** [comment]
I use this adaptor: https://www.facebook.com/ManAdaBue
Viele Grüße von Unterwegs

*Von: *Erik ***@***.***>
*An: *universam1/iSpindel ***@***.***>
*Kopie: *ralfsteck ***@***.***>; Author ***@***.***>
*Datum: *04.11.2023 13:48:29
*Betreff: *Re: [universam1/iSpindel] Idea for bottle manometer based on iSpindel (Discussion #612)

**Comment by ErikdBr** [comment]
After careful considerationt I am going for an analog pressure meter with the swing-top bottle adapter. Just two parts to order and easy to assemble.
Thanks for this tip.

**Comment by ralfsteck** [comment]
Thank you all, but until now we discussed hardware topics and not my real topic: a software fork of the iSpindel firmware for my iBottle. Please, I just need answers to that.

**Reply by ralfsteck** [reply]
who wants to distribute? I plan to open source the project.

**Comment by ralfsteck** [comment]
I don't see replies that I get via mail. strange.

Anyway. I don't want to (or am able to) fork or program anything. I thought the iBottle hardware would be an interesting addition/extension to the iSpindel project.

If not, that's OK for me. I just wanted to help/give back.

**Reply by universam1** [reply]
I have no objection to cherry pick source code of this repo for other purposes, not sure though if it is of huge benefit since the software is tailor fit for the iSpindel. 
I'm happy for further ideas of this ecosystem for sure. If you think iBottle idea fits into this project, and if you like to take the effort, I have no objection.
